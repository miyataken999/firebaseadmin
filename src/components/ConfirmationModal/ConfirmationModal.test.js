import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import ConfirmationModal from '.';

describe('<ConfirmationModal /> rendering', () => {
  const onConfirm = jest.fn();
  const onCancel = jest.fn();

  beforeEach(() => {
    onConfirm.mockClear();
    onCancel.mockClear();
  });

  it('should render without crashing', () => {
    const component = render(
      <ConfirmationModal
        isActive
        confirmButtonMessage="test message"
        onConfirmation={onConfirm}
        cancelButtonMessage="test message"
        onCancel={onCancel}
        title="test title"
        body="test body"
      />
    );

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should set the active modifier if the isActive prop is passed down', () => {
    const component = render(
      <ConfirmationModal
        isActive
        confirmButtonMessage="test message"
        onConfirmation={onConfirm}
        cancelButtonMessage="test message"
        onCancel={onCancel}
        title="test title"
        body="test body"
      />
    );

    expect(
      component.container.querySelector('div.modal.is-active')
    ).toBeTruthy();
  });

  it('should not set the active modifier if the isActive prop is not passed down', () => {
    const component = render(
      <ConfirmationModal
        confirmButtonMessage="test message"
        onConfirmation={onConfirm}
        cancelButtonMessage="test message"
        onCancel={onCancel}
        title="test title"
        body="test body"
      />
    );

    expect(component.container.querySelector('div.modal.is-active')).toBeNull();
  });

  it('should call onConfirm when the confirmation button is clicked', () => {
    const component = render(
      <ConfirmationModal
        isActive
        confirmButtonMessage="confirm test message"
        onConfirmation={onConfirm}
        cancelButtonMessage="test message"
        onCancel={onCancel}
        title="test title"
        body="test body"
      />
    );

    fireEvent.click(component.getByText('confirm test message'));
    expect(onConfirm).toHaveBeenCalled();
  });

  it('should call onCancel when the cancel button is clicked', () => {
    const component = render(
      <ConfirmationModal
        isActive
        confirmButtonMessage="test message"
        onConfirmation={onConfirm}
        cancelButtonMessage="cancel test message"
        onCancel={onCancel}
        title="test title"
        body="test body"
      />
    );

    fireEvent.click(component.getByText('cancel test message'));

    expect(onCancel).toHaveBeenCalled();
  });

  it('should set the title of the modal', () => {
    const component = render(
      <ConfirmationModal
        isActive
        confirmButtonMessage="test message"
        onConfirmation={onConfirm}
        cancelButtonMessage="test message"
        onCancel={onCancel}
        title="test title"
        body="test body"
      />
    );

    expect(component.getByText('test title')).toBeTruthy();
  });

  it('should set the body of the modal', () => {
    const component = render(
      <ConfirmationModal
        isActive
        confirmButtonMessage="test message"
        onConfirmation={onConfirm}
        cancelButtonMessage="test message"
        onCancel={onCancel}
        title="test title"
        body="test body"
      />
    );

    expect(component.getByText('test body')).toBeTruthy();
  });

  it('should set the confirm button message', () => {
    const component = render(
      <ConfirmationModal
        isActive
        confirmButtonMessage="confirm test message"
        onConfirmation={onConfirm}
        cancelButtonMessage="test message"
        onCancel={onCancel}
        title="test title"
        body="test body"
      />
    );

    expect(component.getByText('confirm test message')).toBeTruthy();
  });

  it('should set the cancel button message', () => {
    const component = render(
      <ConfirmationModal
        isActive
        confirmButtonMessage="test message"
        onConfirmation={onConfirm}
        cancelButtonMessage="cancel test message"
        onCancel={onCancel}
        title="test title"
        body="test body"
      />
    );

    expect(component.getByText('cancel test message')).toBeTruthy();
  });
});
