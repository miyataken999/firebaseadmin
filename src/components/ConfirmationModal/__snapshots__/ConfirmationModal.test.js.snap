// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationModal /> rendering should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="modal is-active"
  >
    <div
      class="modal-background"
    />
    <div
      class="modal-card"
    >
      <header
        class="modal-card-head"
      >
        <p
          class="modal-card-title"
        >
          test title
        </p>
      </header>
      <section
        class="modal-card-body"
      >
        test body
      </section>
      <footer
        class="modal-card-foot"
      >
        <button
          class="button is-danger undefined"
          type="button"
        >
          test message
        </button>
        <button
          class="button"
          type="button"
        >
          test message
        </button>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
